qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/AppointmentSteps.java

```diff
@@ -0,0 +1,241 @@
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.contrib.qaframework.automation;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import io.cucumber.java.After;
+import io.cucumber.java.Before;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+
+import org.openmrs.contrib.qaframework.RunTest;
+import org.openmrs.contrib.qaframework.helper.TestData;
+import org.openmrs.contrib.qaframework.page.AppointmentBlocksPage;
+import org.openmrs.contrib.qaframework.page.AppointmentSchedulingPage;
+import org.openmrs.contrib.qaframework.page.ManageAppointmentsPage;
+import org.openmrs.contrib.qaframework.page.ManageProviderSchedulesPage;
+import org.openmrs.contrib.qaframework.page.RequestAppointmentPage;
+
+public class AppointmentSteps extends Steps {
+
+	private AppointmentSchedulingPage appointmentSchedulingPage;
+	private AppointmentBlocksPage appointmentBlocksPage;
+	private ManageAppointmentsPage manageAppointmentsPage;
+	private ManageProviderSchedulesPage manageProviderSchedulesPage;
+	private RequestAppointmentPage requestAppointmentPage;
+	private TestData.PatientInfo testPatient;
+	private static final String FIRST_SERVICE_NAME = "Oncology";
+	private static final String SECOND_SERVICE_NAME = "Dermatology";
+	private static final String LOCATION = "Laboratory";
+
+	@Before(RunTest.HOOK.SELENIUM_APPOINTMENT)
+	public void visitDashboard() {
+		initiateWithLogin();
+		testPatient = createTestPatient();
+		findPatientPage = homePage.goToFindPatientRecord();
+		findPatientPage.enterPatient(testPatient.identifier);
+		findPatientPage.waitForPageToLoad();
+		dashboardPage = findPatientPage.clickOnFirstPatient();
+	}
+
+	@After(RunTest.HOOK.SELENIUM_APPOINTMENT)
+	public void destroy() {
+		deletePatient(testPatient);
+		quit();
+	}
+
+	@Given("a user clicks on Request Appointment link from the Patient dashboard")
+	public void loadRequestAppointmentPage() {
+		requestAppointmentPage = (RequestAppointmentPage) dashboardPage.clickOnRequest();
+	}
+
+	@When("the system loads Request Appointment page")
+	public void systemLoadsRequestAppointmentPage() {
+		assertTrue(textExists("Request Appointment"));
+	}
+
+	@And("a user fills in the appointment request details")
+	public void captureAppointmentRequestDetails() {
+		requestAppointmentPage.enterAppointmentType(FIRST_SERVICE_NAME);
+		requestAppointmentPage.enterMinimumValue("0");
+		requestAppointmentPage.selectMinimumUnits("Day(s)");
+		requestAppointmentPage.enterMaximumValue("2");
+		requestAppointmentPage.selectMaximumUnits("Day(s)");
+		requestAppointmentPage.addAppointmentRequestNote("I need to see a doctor first");
+	}
+
+	@And("a user clicks save button")
+	public void saveAppointmentRequest() {
```

Comment:
```suggestion
	public void userClicksOnSaveAppointmentRequest() {
```
What do you think about the method name here.

Reply:
> What do you think about the method name here.

@sherrif10 in my opinion, if I changed this method to `userClicksOnSaveAppointmentRequest` it looks like there's an element or button that contains the word SaveAppointmentRequest. In this case I would think, it would be better to go with `saveAppointmentRequest` .